From 13c051f973bde459c1e9fc80b46a11e803058b06 Mon Sep 17 00:00:00 2001
From: Mike Gerdts <mike.gerdts@joyent.com>
Date: Tue, 15 Oct 2019 14:53:40 -0500
Subject: [PATCH] TOOLS-2336 eng repo should draw attention to build zone
 instructions Reviewed by: Tim Foster <tim.foster@joyent.com> Approved by: Tim
 Foster <tim.foster@joyent.com>

---
 Makefile                   |  2 +-
 README.md                  |  8 +++++++-
 tools/validate-buildenv.sh | 12 ++++++++++++
 3 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index e0b5eef..938550f 100644
--- a/Makefile
+++ b/Makefile
@@ -202,7 +202,7 @@ RELEASE_TARBALL = $(NAME)-pkg-$(STAMP).tar.gz
 #
 
 # This image is triton-origin-x86_64-18.4.0
-BASE_IMAGE_UUID = a9368831-958e-432d-a031-f8ce6768d1
+BASE_IMAGE_UUID = a9368831-958e-432d-a031-f8ce6768d190
 BUILDIMAGE_NAME = manta-myproject
 BUILDIMAGE_DESC	= My Project Has A Description
 AGENTS          = amon config registrar
diff --git a/README.md b/README.md
index 61f2a33..e125a04 100644
--- a/README.md
+++ b/README.md
@@ -5,7 +5,7 @@
 -->
 
 <!--
-    Copyright (c) 2016, Joyent, Inc.
+    Copyright 2019 Joyent, Inc.
 -->
 
 # eng: Joyent Engineering Guide
@@ -43,6 +43,12 @@ After the boilerplate paragraph, write a brief description about your repo.
 
 ## Development
 
+To ensure maximum compatibility, release builds are performed on a build zone
+that is old enough to allow new and updated components to run on all supported
+platform images.  If you are not using the Joyent Jenkins instance for
+performing builds, you should build using an appropriate build zone.  See
+[Build Zone Setup for Manta and Triton](https://github.com/joyent/triton/blob/master/docs/developer-guide/build-zone-setup.md).
+
 Describe steps necessary for development here.
 
     make all
diff --git a/tools/validate-buildenv.sh b/tools/validate-buildenv.sh
index a3c4afb..8eb2351 100755
--- a/tools/validate-buildenv.sh
+++ b/tools/validate-buildenv.sh
@@ -13,6 +13,11 @@
 # Check if the current build machine is supported for building this component
 # and that the build environment seems sane.
 #
+# Most Joyent software is built using a common build environment described in
+# https://github.com/joyent/triton/blob/master/docs/developer-guide/build-zone-setup.md.
+# The quickest path to having a sane build environment likely involves following
+# that document.
+#
 # Ideally, rather than checking the sanity of any random shell environment,
 # we'd have a way to fully specify the build environment, sanitizing it so that
 # only specific user-set environment variables are allowed. That is not done
@@ -751,6 +756,13 @@ else
     # bits-upload not posting to Manta/updates.joyent.com.
     verify_clean_repo
     if [[ "$RESULT" -gt 0 ]]; then
+        echo ""
+        echo "Build zone setup typically requires almost no work if you are"
+        echo "using the right image.  See:"
+        echo ""
+        echo "https://github.com/joyent/triton/blob/master/docs/developer-guide/build-zone-setup.md"
+        echo ""
+
         exit 1
     else
         exit 0
-- 
2.17.2 (Apple Git-113)

